schema/ext/vulnerability-2.0-SNAPSHOT.schema.json

```diff
@@ -0,0 +1,235 @@
+                2.9,
+                7.2
+              ]
+            },
+            "exploitability": {
+              "type": "number",
+              "title": "Exploitability Score",
+              "description": "The exploitability subscore of the security vulnerability (Refer CVSS standard for example)",
+              "multipleOf": 0.1,
+              "minimum": 0,
+              "maximum": 10,
+              "examples": [
+                2.9,
+                7.2
+              ]
+            }
+          }
+        }
+      ]
+    },
+    "rating": {
+      "type": "object",
+      "title": "Rating",
+      "description": "Defines the risk ratings of a vulnerability.",
+      "properties": {
+        "score": {
+          "$ref": "#/definitions/score"
+        },
+        "severity": {
+          "$ref": "#/definitions/severity"
+        },
+        "method": {
+          "$ref": "#/definitions/ratingMethod"
+        },
+        "source": {
+          "type": "string",
+          "title": "Source",
+          "description": "The entity responsible for providing the rating",
+          "examples": [
+            "NVD"
+          ]
+        },
+        "vector": {
+          "type": "string",
+          "title": "Vector",
+          "description": "Textual representation of the metric values used to score the vulnerability see attack vector in https://www.first.org/cvss/v3.1/specification-document"
+        }
+      }
+    },
+    "source": {
+      "type": "object",
+      "title": "Source",
+      "description": "The source of the vulnerability where it is documented. Usually the name of the organization publishing vulnerability information",
+      "properties": {
+        "url": {
+          "type": "string",
+          "title": "URL",
+          "description": "The url of the vulnerability documentation as provided by the source.",
+          "examples": [
+            "https://nvd.nist.gov/vuln/detail/CVE-2019-15842"
+          ]
+        },
+        "name": {
+          "type": "string",
+          "title": "Name",
+          "description": "The name of the source.",
+          "examples": [
+            "NVD",
+            "National Vulnerability Database",
+            "OSS Index",
+            "VulnDB",
+            "NPM Advisories"
+          ]
+        }
+      }
+    },
+    "vulnerability": {
+      "type": "object",
+      "title": "Vulnerability",
+      "description": "Defines the structure of a vulnerability.",
+      "properties": {
+        "ref": {
+          "type": "string",
+          "format": "string",
+          "title": "Reference",
+          "description": "References a component by the components bom-ref attribute"
+        },
+        "id": {
+          "type": "string",
+          "title": "ID",
+          "description": "The id of the vulnerability as defined by the risk scoring methodology. For example CVE-2019-15842 (of https://nvd.nist.gov/vuln/detail/CVE-2019-15842)"
+        },
```

Comment:
@garethr I'm curious why you have a single ID, but have multiple sources.  Wouldn't each source potentially have its own ID?

CVE-xxxx-xxxx
   aka
SNYK-xxxxxxxxx
   aka
SONATYPE-xxxxxxxx

Reply:
Also, wouldn't it be possible that each source could potentially have slightly different ratings as well? I'm thinking of VulnDB for example which makes a ton of corrections and enhancements to the NVD and its not uncommon for them to provide both the NVD scores as well as their own.

Are you trying to capture the ability to specify that a vulnerability will be known by different names across different sources?